src/themes/default/elements/segment.variables

```diff
@@ -132,21 +131,9 @@
 @tertiaryColor: @mutedTextColor;
 
 @secondaryInvertedLightness: 0.2;
-@secondaryInvertedBackground:
-    lighten(@black, (@secondaryInvertedLightness * 100))
-    linear-gradient(
-        rgba(255, 255, 255, @secondaryInvertedLightness) 0,
-        rgba(255, 255, 255, @secondaryInvertedLightness) 100%
-    )
-;
+@secondaryInvertedBackground: lighten(@black, (@secondaryInvertedLightness * 100)) linear-gradient(rgba(255, 255, 255, @secondaryInvertedLightness) 0, rgba(255, 255, 255, @secondaryInvertedLightness) 100%);
```

Comment:
same here: we should ignore/keep function parameters lists in separates lines just like for usual lists (box-shadow, etc)

Reply:
impossible, no ignore in file supported, the only way is to patch Prettier but I am not sure if such patch will ever be merged into Prettier directly...

Reply:
> impossible, no ignore in file supported, the only way is to patch Prettier but I am not sure if such patch will ever be merged into Prettier directly...

Another reason to create a fork which does properly what we want :wink:

Reply:
It is all about 1 file - https://github.com/prettier/prettier/blob/main/src/language-css/printer-postcss.js

I would still prefer no fork and replace the CSS/LESS printer only.

All the required files should be exported - https://github.com/prettier/prettier/blob/main/src/language-css/index.js - so it should be possible to create custom printer that will fallback to the native one. I guess it can be done in ~100 lines of JS code. But not easy ones to find out :)

I can look into it quickly, but maybe we can be fine with one extra patch of function args to be on separate like when separated with comma.

Reply:
I tried a print-width of 0 and for this case the result is more readable in a whole, yet more lines, but more consistent.
It's still not perfect ( the * 100 is also put into a new line)
However, i did not try the whole code, just a quick test on

https://prettier.io/playground/#N4Igxg9gdgLgprEAuEABAznSUAmBDAJwE8BJKANzgPhwCE8wBrAcwIgFdckAdKAAgF8ANgEtmAC3hQAFKgBGQhowA0fWZmz5iZStTg4AMmMlQ46dHwBUfAIwAGOwEpHvQcJGnCAWlZ4cIhBhpVzdBAmY5PGkAJgBWWNU4hL4k1QwsaC1SCioaIwkYU3NHPjtlENC+cMiY+MS6lIb0zUJs3TzjQrN0Evs7AFIKvhcoAG4QZRAIAAcYEWh0ZFBCNgB3AAVCBEWUPCFVvCJFybkCJTgYAGU8AFs4I1NkADM9zBOzpgvL6YYPZmQYAR2HBJnAbnJ9Dh9AY8FBmOw8Mw4AAxCAEG54GBzOHIEB4dgwCATECSG5CADq4hE8HQPzAcEu22pInI1KIuLA5mJHkw1HWZ2YGOerxBIAAVugAB6XP5COAARXYEHgwqEbxAPwIvNxcq5k2mBA8MHJIhwMHEyDKGrYmHJZ2muINZiolGJAEclfB+TMdnj0F5TPp9MSCHAPSJQ-zEUKkC81aLMDcRACgQnZXAAIJYw1yAlwdZUB5wVXq9DpxXK4uxkWTGB4OQms0WpDRWtnESiOEAYQgNxjIDMsWJ7EwABV6zs4+ryMCyFDYJcwIbZhncJcYEQ5SW4ABfHdAA
![image](https://user-images.githubusercontent.com/18379884/208324318-dbc1f028-453e-4849-b536-8901f95e69ed.png)

If there only would be a possiblity to have comment hints to ignore some lines which are already perfect (for us, not what prettiert considers perfect)

Reply:
👍 for the finding, but as you saw, this wraps the selectors too...

can you propose a prettier patch against my branch?

Reply:
I spent several hours in this. The best place to patch is https://github.com/prettier/prettier/blob/2.8.1/src/language-css/printer-postcss.js#L840

where something like:
```
        if (parentParentNode.type === "value-root") {
          parts.push(hardline);
          didBreak = true;

          continue;
        }
```

can be added, then all values separated by spaces are placed on new lines like:

![image](https://user-images.githubusercontent.com/2228672/208411751-bd1a2942-9689-4aa3-87a4-5fc9756ad8fa.png)

but math operations (not inside parentheses) are too:

![image](https://user-images.githubusercontent.com/2228672/208411832-f68d8709-a122-4892-9607-a9a4408832f1.png)

and imports:

![image](https://user-images.githubusercontent.com/2228672/208411947-14d376eb-ee87-4222-9f38-f7de85f7ebfa.png)

the imports can be fixed with ` && declAncestorProp` condition but then variable assignments are unfixed (as for ex. `margin` can be assigned to any property name, so we are short on data)

I also tried ` && iNextNode.raws.before.includes("\n")` condition to check if there was a new line originally, but there are only like 15 places where applicable (or not possible to patch on one place) as other places/AST types look like:

![image](https://user-images.githubusercontent.com/2228672/208415131-41a301f5-61f5-4d3e-90cc-4f7486c1e5b3.png)

but once we start to fix function argumnets, should be wrap simple/short `rgb` arguments to? no, we should not

so I would say we cannot do much and I would prefer to get this PR merged with maxWidth=Infinity which fixes some unlogical wrappings

actually, if we look on https://github.com/fomantic/Fomantic-UI/commit/470cd4e7b1dbce900d5e006aa8f3c5d7bf850009, there are not so many places "that would be better formatted differently", so I guess this is acceptable vs. the gain with Prettier integrated

Reply:
I am honest with you here and i _really_ appreciate all the time you spent into this:
In the current stage, i don't want to merge the PR. It is not satisfying as a whole and that manual patching is too hacky. That's not the way we should implement/use an external package.

Let's see if/when your proposed PRs in the prettier repo get merged and what could be patched as well to get that "linear-gradient()/functions-parameter-each-line (but not rgb as you said)"-formatting could be improved.

A general possibility for comment-hints like in stylelint or eslint would solve it as we simply could define where prettier should not touch the existing code (instead of figuring out some complex "always to use" formatting syntax)

The current formatted code is IMHO much better or "prettier enough" than what prettier makes out of it.

And, in the end, all it does is doing/expecting whitespace (which you already did in the previous PRs) which gives much less benefit to the CI than what you did with eslint/stylelint already.

However, many changes in this PR i would actually agree with and some of them actually fixes errors (like missing or misplaced semicolons), but for many others i am not sure if really like the change (like the list items per line in the variation.variables file) while the "put everything in one line" is either not consistent all the time and is not really "prettier" than before. (yes, that's of course opinion based 😉 but i would like to concentrate again on improving/fixing FUI components instead of discussing whitespace rules.

Reply:
there are 4 patches currently - 3 regular Prettier bugfixes, 1 patch to adjust the CS as much as possible to the current CS, the first 3 bugfixes were already approved and should be in the next Prettier release

I agree we should focus on FUI other than whitespaces, but I would prefer this topic to be finished, otherwise we will need to focus again, and be in the same exact point

the original review comments are for maxWidth=120, can you remove them, and add comments for current maxWidth=Infinity, ideally only once per CS problem, maybe with number of importance/occurrences, and how you propose it to be fixed by AST

I can then try to fix the problems with rules based on the property/variable name, something that will never be accepted into Prettier, but we can patch it (and possibly create a fork for it)

Reply:
> the original review comments are for maxWidth=120, can you remove them, and add comments for current maxWidth=Infinity, ideally only once per CS problem

Yes, but i was not aware of the consequences as i did not use prettier before. (And when i look at that many issues/open PRs at prettiert,i feeld many people are not completely satisfied with it and forked it to their needs (mostly JS related, but anyway)

> I can then try to fix the problems with rules based on the property/variable name, something that will never be accepted into Prettier, but we can patch it (and possibly create a Prettier fork)

If you like, i can fork it to the fomantic organization and give you rights to maintain it. We eventually could then publish it to npm and fomantic-scope to have a "real" package to rely on in CI.